cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -760,12 +766,16 @@ public MapAndStatus(T loadMap, K status) {
                                     vertx.cancelTimer(t);
                                     p.complete(buildRebalanceStatus(null, KafkaRebalanceState.Stopped, validate(reconciliation, currentKafkaRebalance)));
                                 } else {
-
                                     requestRebalance(reconciliation, host, apiClient, kafkaRebalance, true, rebalanceOptionsBuilder,
                                             currentKafkaRebalance.getStatus().getSessionId())
                                         .onSuccess(rebalanceMapAndStatus -> {
                                             // If the returned status has an optimization result then the rebalance proposal
                                             // is ready, so stop the polling
+                                            KafkaRebalanceStatus status = rebalanceMapAndStatus.getStatus();
+                                            if (rebalanceAnnotation(reconciliation, currentKafkaRebalance) != KafkaRebalanceAnnotation.none) {
+                                                status.addCondition(StatusUtils.buildWarningCondition("InvalidAnnotation", "Wrong annotation value: " + kafkaRebalance.getMetadata().getAnnotations() + ". The rebalance resource supports the following annotations in 'PendingProposal' state: approve, refresh, stop."));
```

Comment:
I think the error message is wrong. We don't support `approve` and `refresh` annotations on a `ProposalPending` state? It think we only support `stop`, at least that is what the code suggests? 

Also, can you split this over a few lines to make it easier to read on thinner screens.